jukebox_radio/streams/models/queue_interval_model.py

```diff
@@ -57,52 +65,149 @@ def create_queue_interval(
+            )
+            .order_by('upper_bound__timestamp_ms')
+            .last()
         )
 
+        if not highest_interval:
+            return False
+
+        lower_bound_limit = highest_interval.upper_bound.timestamp_ms
+        if proposed_lower_bound.timestamp_ms < lower_bound_limit:
+            return True
+
+        return False
+
     def conflicting_interval(self, *, queue_id, lower_bound_id, upper_bound_id):
+        """
+        This method is called before creating a new queue interval between two
+        given bounds. It returns a boolean signifying wether there is a
+        conflict.
+          - True: there is a conflict - DO NOT create queue interval.
+          - False: there is no conflict - ok to create queue interval.
+        """
         Marker = apps.get_model('streams', 'Marker')
         Queue = apps.get_model('streams', 'Queue')
+        QueueInterval = apps.get_model('streams', 'QueueInterval')
 
         queue = Queue.objects.select_related("track").get(uuid=queue_id)
         track = queue.track
 
         lower_bound = Marker.objects.get(uuid=lower_bound_id, track=track)
         upper_bound = Marker.objects.get(uuid=upper_bound_id, track=track)
 
-        return self.filter(
-            Q(
+        return (
+            QueueInterval
+            .objects
+            .filter(
                 Q(
-                    queue_id=queue_id,
-                    lower_bound__timestamp_ms__lt=upper_bound.timestamp_ms,
-                    lower_bound__timestamp_ms__gt=lower_bound.timestamp_ms,
-                ) |
-                Q(
-                    queue_id=queue_id,
-                    upper_bound__timestamp_ms__gt=lower_bound.timestamp_ms,
-                    upper_bound__timestamp_ms__lt=upper_bound.timestamp_ms,
-                ) |
-                Q(
-                    queue_id=queue_id,
-                    lower_bound__timestamp_ms=lower_bound.timestamp_ms,
-                    upper_bound__timestamp_ms=upper_bound.timestamp_ms,
+                    Q(
+                        queue_id=queue_id,
+                        lower_bound__timestamp_ms__lt=upper_bound.timestamp_ms,
+                        lower_bound__timestamp_ms__gt=lower_bound.timestamp_ms,
+                        deleted_at__isnull=True,
+                    ) |
+                    Q(
+                        queue_id=queue_id,
+                        upper_bound__timestamp_ms__gt=lower_bound.timestamp_ms,
+                        upper_bound__timestamp_ms__lt=upper_bound.timestamp_ms,
+                        deleted_at__isnull=True,
+                    ) |
+                    Q(
+                        queue_id=queue_id,
+                        lower_bound__timestamp_ms=lower_bound.timestamp_ms,
+                        upper_bound__timestamp_ms=upper_bound.timestamp_ms,
+                        deleted_at__isnull=True,
+                    )
                 )
             )
+            .exists()
```

Comment:
I updated all callers of this method to accommodate for the new return value type (prev: queryset, now: bool)